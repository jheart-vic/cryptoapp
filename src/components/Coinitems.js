import React from 'react';
import PropTypes from 'prop-types';
import { FiArrowRightCircle } from 'react-icons/fi';
import './Coin.css';
import { Link } from 'react-router-dom';

const CoinItems = ({ coin }) => {
  const {
    symbol, name, ath, image, id,
  } = coin;
  return (
    <div>
      <div className="parent">
        <div>
          <Link className="Links" to={`/CoinDetails/${id}`} state={{ coin, id }}>
            <div className="icon">
              <FiArrowRightCircle />
            </div>
          </Link>
          <div className="title">
            <h2 className="Name">{name}</h2>
          </div>
          <div className="image">
            <img src={image} alt="coinImages" className="img" />
          </div>
          <div className="div">
            <p className="ath">{ath}</p>
            <p className="Symbol">{symbol}</p>
          </div>
        </div>
      </div>
    </div>
  );
};

CoinItems.propTypes = {
  coin: PropTypes.shape({
    name: PropTypes.string.isRequired,
    symbol: PropTypes.string.isRequired,
    ath: PropTypes.number.isRequired,
    image: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
  }).isRequired,
};
export default CoinItems;
