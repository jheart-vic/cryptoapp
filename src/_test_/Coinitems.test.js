import React from 'react';
import '@testing-library/jest-dom';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import Coinitems from '../components/Home'
import store from '../redux/configureStore';

it('renders correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Coinitems />
      </Provider>,
    ).toJSON();
  expect(tree).toMatchSnapshot();
});