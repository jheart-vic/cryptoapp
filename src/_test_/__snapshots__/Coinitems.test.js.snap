// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="wrapper"
>
  <div>
    <div
      className="carousel slide carousel-fade"
      onKeyDown={[Function]}
      onMouseOut={[Function]}
      onMouseOver={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
    >
      <div
        className="carousel-indicators"
      >
        <button
          aria-current={true}
          aria-label="Slide 1"
          className="active"
          data-bs-target=""
          onClick={[Function]}
          type="button"
        />
        <button
          aria-current={false}
          aria-label="Slide 2"
          data-bs-target=""
          onClick={[Function]}
          type="button"
        />
        <button
          aria-current={false}
          aria-label="Slide 3"
          data-bs-target=""
          onClick={[Function]}
          type="button"
        />
      </div>
      <div
        className="carousel-inner"
      >
        <div
          className="active carousel-item"
        >
          <img
            alt="Market ratings"
            className="d-block w-100"
            src="Crypto1.jpg"
          />
          <div
            className="carousel-caption"
          >
            <h3>
              This is a beautiful image
            </h3>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="Ratings"
            className="d-block w-100"
            src="Crypto2.jpg"
          />
          <div
            className="carousel-caption"
          >
            <h3>
              This is another most fancy a
            </h3>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
            </p>
          </div>
        </div>
        <div
          className="carousel-item"
        >
          <img
            alt="coin ratings"
            className="d-block w-100"
            src="Crypto3.jpg"
          />
          <div
            className="carousel-caption"
          >
            <h3>
              This is another Beautiful Image
            </h3>
            <p>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
            </p>
          </div>
        </div>
      </div>
      <a
        className="carousel-control-prev"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          aria-hidden="true"
          className="carousel-control-prev-icon"
        />
        <span
          className="visually-hidden"
        >
          Previous
        </span>
      </a>
      <a
        className="carousel-control-next"
        href="#"
        onClick={[Function]}
        onKeyDown={[Function]}
        role="button"
        tabIndex={0}
      >
        <span
          aria-hidden="true"
          className="carousel-control-next-icon"
        />
        <span
          className="visually-hidden"
        >
          Next
        </span>
      </a>
    </div>
  </div>
  <div>
    <div
      className="search"
    >
      <div
        className="center input-group input-group-lg"
        onChange={[Function]}
      >
        <span
          className="input-group-text"
          id="inputGroup-sizing-lg"
        >
          Search
        </span>
        <input
          aria-describedby="inputGroup-sizing-sm"
          aria-label="Large"
          className="form-control"
        />
      </div>
    </div>
    <div
      className="grid"
    />
  </div>
</div>
`;
